code/game/turfs/turf.dm

```diff
@@ -252,6 +252,16 @@ GLOBAL_LIST_EMPTY(created_baseturf_lists)
 			return 1
 	return 0
 
+/// Returns TRUE if the turf is occupied by a big object already (i.e. computer)
+/proc/is_occupied_turf(turf/T, exclude_mobs=TRUE)
+	if(is_blocked_turf(T, exclude_mobs)) // check walls first
+		return TRUE
+	for(var/obj/O in T)
+		if(O.layer >= BELOW_OPEN_DOOR_LAYER) // something big occupied the turf
```

Comment:
just check if there is no obj/structure or turf/closed
layer stuff is not reliable at all

Reply:
Windows aren't closed turf so it won't work

Reply:
![image](https://user-images.githubusercontent.com/87972842/205206659-d1452b70-5fa8-41cf-9218-217078af429e.png)

windows and some stuffs aren't closed turf so it won't work.

Reply:
`density` determines if the object is passable or not. Check if there is any /obj with density and skip the turf.

Reply:
That is is_blocked_turf() proc inside.

Reply:
ah nevermind. I was sleepy and misunderstood what you meant